src/cpp/src/whisper/pipeline_static.cpp

```diff
@@ -393,20 +398,130 @@ void add_attention_mask_input(std::shared_ptr<ov::Model> model) {
+                if (ov::is_type<ov::op::v13::ScaledDotProductAttention>(node)) {
+                    if (node->inputs().size() == 4u) {
+                        self_attn_nodes.push_back(node);
+                    } else {
+                        cross_attn_nodes.push_back(node);
+                    }
+                }
+            }
+
+            // Self-attention
+            OPENVINO_ASSERT(!self_attn_nodes.empty());
+
+            auto attention_mask = std::make_shared<v0::Parameter>(ov::element::i64, ov::PartialShape{-1, -1});
+            attention_mask->get_output_tensor(0).set_names({"attention_mask"});
+            model->add_parameters({attention_mask});
+
+            const auto kAttnMaskPort = 3;
+            auto slice = self_attn_nodes[0]->input(kAttnMaskPort).get_source_output().get_node();
+            auto cvt = std::make_shared<v0::Convert>(attention_mask->output(0), ov::element::f32);
+            auto add = std::make_shared<v1::Add>(slice->output(0), cvt->output(0));
+
+            auto trps = std::make_shared<v1::Transpose>(cvt->output(0), v0::Constant::create(ov::element::i32, ov::Shape{2}, std::vector<int>{1, 0}));
+            auto mtpl = std::make_shared<v1::Multiply>(trps->output(0), add->output(0));
+
+            auto cst_ninf = std::make_shared<v0::Constant>(
+                ov::element::f32,
+                ov::Shape{1},
+                std::vector<float>{-std::numeric_limits<float>::max()}
+            );
+            auto cst_1 = std::make_shared<v0::Constant>(
+                ov::element::f32,
+                ov::Shape{1},
+                std::vector<float>{1}
+            );
+            auto cst_0 = std::make_shared<v0::Constant>(
+                ov::element::f32,
+                ov::Shape{1},
+                std::vector<float>{0}
+            );
+
+            auto equal = std::make_shared<v1::Equal>(mtpl->output(0), cst_1->output(0));
+            auto select = std::make_shared<v1::Select>(
+                equal->output(0), cst_0->output(0), cst_ninf->output(0)
+            );
+
+            for (auto self_attn : self_attn_nodes) {
+                self_attn->input(3).replace_source_output(select->output(0));
+            }
+
+            if (transform_cross_attn) {
+                // Cross attn
+                OPENVINO_ASSERT(!cross_attn_nodes.empty());
+                // FIXME: Should be taken from topology - don't hardcode!!!
+                auto shape_cst = std::make_shared<v0::Constant>(
+                    ov::element::i64,
+                    ov::Shape{2},
+                    std::vector<float>{MAX_PROMPT_LEN, 1}
+                );
+                // FIXME: Avoid "1500" hardcode!!!
+                auto target_shape = std::make_shared<v0::Constant>(
+                    ov::element::i64,
+                    ov::Shape{2},
+                    std::vector<float>{MAX_PROMPT_LEN, 1500}
+                );
+                // FIXME: Must be transpose if batch present
```

Comment:
Is it going to be resolved in this PR or is it a non-blocker?

Reply:
It's not a blocker as NPU doesn't support now batch size >1, there's a check for this and resize in function full_decode()